### PR TITLE
feat(ram): add a data source to query the list of RAM shared principals

### PR DESCRIPTION
**Pull request by zhangting951007**
**What this PR does / why we need it**:
+ add data source to query the lis of RAM shared principals

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
+ The query method is POST, so use a for loop to query。

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/ram" TESTARGS="-run TestAccDatasourceRAMSharedPrincipals_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ram -v -run TestAccDatasourceRAMSharedPrincipals_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceRAMSharedPrincipals_basic
=== PAUSE TestAccDatasourceRAMSharedPrincipals_basic
=== CONT  TestAccDatasourceRAMSharedPrincipals_basic
--- PASS: TestAccDatasourceRAMSharedPrincipals_basic (25.92s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ram       25.984s
```

### CODE REVIEW

File: docs/data-sources/ram_shared_principals.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `resource_owner` - (Required, String) Specifies the owner associated with the RAM share.
+  Value options are as follows:
+    + **self**: Shared to other users by myself.
```
Comment (deer-hang) [inline docs/data-sources/ram_shared_principals.md]: Leave two spaces in front.

File: huaweicloud/services/acceptance/ram/data_source_huaweicloud_ram_shared_principals_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  resource_share_id = data.huaweicloud_ram_shared_principals.test.shared_principals.0.resource_share_id
+}
+
+output "resource_share_id_filter_is_useful" {
+  value = length(data.huaweicloud_ram_shared_principals.test.shared_principals) > 0 && alltrue(
+    [for v in data.huaweicloud_ram_shared_principals.resource_share_id_filter.shared_principals[*].resource_share_id : v == local.resource_share_id]
+  )
+}
+
+
```
Comment (jinyangyang222) [inline huaweicloud/services/acceptance/ram/data_source_huaweicloud_ram_shared_principals_test.go:76]: Delete unnecessary blank lines

File: huaweicloud/services/acceptance/ram/data_source_huaweicloud_ram_shared_principals_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	rName := "data.huaweicloud_ram_shared_principals.test"
+	principalTestname := "data.huaweicloud_ram_shared_principals.principal_filter"
+	resourceURNTestName := "data.huaweicloud_ram_shared_principals.resource_urn_filter"
+	dc := acceptance.InitDataSourceCheck(rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckRAMSharedPrincipals(t)
+			acceptance.TestAccPreCheckRAM(t)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/ram/data_source_huaweicloud_ram_shared_principals_test.go]: Please write a separate preCheck. There are two unnecessary parameters in the current preCheck.

File: huaweicloud/services/ram/data_source_huaweicloud_ram_shared_pricipals.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+	bodyParams := map[string]interface{}{
+		"resource_urn":   utils.ValueIngoreEmpty(d.Get("resource_urn")),
+		"resource_owner": d.Get("resource_owner"),
+		"limit":          pageLimit,
+	}
+
+	principals := make([]interface{}, 0)
+	if v, ok := d.GetOk("principal"); ok {
+		principals = append(principals, utils.ValueIngoreEmpty(v))
+		bodyParams["principals"] = principals
```
Comment (deer-hang) [inline huaweicloud/services/ram/data_source_huaweicloud_ram_shared_pricipals.go]:
```suggestion
		bodyParams["principals"] = []interface{}{v}
```
Same to `resource_share_ids`

**Comment by zhangting951007** [comment]
```
make testacc TEST="./huaweicloud/services/acceptance/ram" TESTARGS="-run TestAccDatasourceRAMSharedPrincipals_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ram -v -run TestAccDatasourceRAMSharedPrincipals_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceRAMSharedPrincipals_basic
=== PAUSE TestAccDatasourceRAMSharedPrincipals_basic
=== CONT  TestAccDatasourceRAMSharedPrincipals_basic
--- PASS: TestAccDatasourceRAMSharedPrincipals_basic (24.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ram       24.053s
```

File: docs/data-sources/ram_shared_principals.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  + **self**: Shared to other users by myself.
+  + **other-accounts**: Shared to me by other users.
+
+* `principal` - (Optional, String) Specifies the principal associated with the RAM share.
+  The principal could be account ID or organization ID.
+  + If set to account ID, please make sure the account ID is not your owner account ID.
+  + If set to organization ID, you first need to use the RAM console to enable sharing with Organization. Please refer
+  to the [document](https://support.huaweicloud.com/intl/en-us/qs-ram/ram_02_0004.html).
+
+* `resource_share_id` - (Optional, String) Specifies the ID of resource share.
```
Comment (jinyangyang222) [inline docs/data-sources/ram_shared_principals.md:35]: The parameter order is inconsistent with the schema

**Pull request merged**